src/App.tsx

```diff
@@ -1,16 +1,24 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movieList, setMovielist] = useState<Movie[]>(moviesFromServer);
+
+  const addMovies = (newMovie: Movie) => {
+    setMovielist((currentMovies) => [...currentMovies, newMovie]);
+  };
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movieList} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={addMovies} />
```

Comment:
```suggestion
        <NewMovie onAdd={addMovies} key={count} />
```
Move your counter from NewMovie in App component. When you change counter - it will create form from scratch.